### ISSUE TITLE
Regression: Pyright treats container fields holding Pydantic models as partially unknown

### ISSUE THREAD

**Issue by mpaluch92** [microsoft/pyright#10442]
**Describe the bug**
Starting with **Pyright 1.1.399**, container fields (e.g., `list`, `set`) that hold Pydantic models are incorrectly reported as having partially unknown types.

This issue did **not** occur in earlier versions of Pyright. The type of these fields should be inferred correctly, as it was before.

Used versions:
- Pyright 1.1.399 or later
- Pydantic 2.11.4
- Python 3.12.7

**Code To Reproduce**

```python
from pydantic import BaseModel, Field


class User(BaseModel):
    name: str = Field(default=..., description="Name of the user")

class UserCollection(BaseModel):
    users: list[User] = Field(default_factory=list, description="List of users")
```

**Pyright Output**:

```
error: Type of "users" is partially unknown
    Type of "users" is "list[Unknown]" (reportUnknownVariableType)
```

Similarly, same holds true for Pydantic `SecretStr`:

```python
from pydantic import BaseModel, Field, SecretStr


class User(BaseModel):
    passwords: list[SecretStr] = Field(
        default_factory=list, description="List of passwords for the user"
    )
```

with Pyright complaining that:

```
error: Type of "passwords" is partially unknown
    Type of "passwords" is "list[Unknown]" (reportUnknownVariableType)
```
